### ISSUE TITLE
Build command creates invalid zip files under Windows 10

### ISSUE THREAD

**Issue by davidski** [apex/apex#596]
Environment
Windows 10 x64
apex version 0.10.3

Behavior: When running `apex build simple > out.zip`, the resulting zip file is corrupt according to both Windows Explorer and the AWS Lambda console.

Expected behavior: A valid zip file is generated. :smile:

Configuration: 
project.json
```
{
  "name": "python",
  "description": "Python example project",
  "role": "arn:aws:iam::293503197324:role/apex_lambda_function"
}
```

functions\simple\main.py
```
print "start simple function"
def handle(event, context):
    print "processing event"
return event
```

Debug output:
```
D:\apex-text> apex build simple --log-level debug > out.zip
   • loading functions in functions
   • open                      env= function=simple
   • inferring runtime         env= function=simple
   • inferred runtime          env= function=simple runtime=python
   • creating build            env= function=simple
   • injecting prelude         env= function=simple
   • adding env                env=map[APEX_FUNCTION_NAME:simple LAMBDA_FUNCTION_NAME:python_simple] function=simple
   • add file to zip           env= file=main.py function=simple
   • add file to zip           env= file=main.py.bak function=simple
   • conditional flush         above_duration=24h0m0s above_size=500 age=57m44.565296s size=6
   • close
```

**Comment by davidski** [comment]
Updating to note that this behavior still persists under the new 0.11.0 release, but x32 and x64 Windows versions.

**Comment by tj** [comment]
I'm no help with Windows sorry! Happy to look at a PR if anyone has a fix

**Comment by davidski** [comment]
A few notes for anyone able to take a look at this. 

- Deploy commands work fine.
- Build commands piping to a zip file generate a corrupt zip when running under `powershell.exe` as a command shell. Both `> test.zip` and `| Output-File test.zip` syntax versions fail in similar fashion.
- Build commands under `cmd.exe` work as expected.
- The failure is reproducible across multiple Windows 10 installs.

This is the darndest thing -- I'm not aware what could be causing STDOUT to behave differently across the different shells. I've done some light hacking on a different `archive\zip` implementation that works fine to a file, but I'm not yet familiar enough with golang to grok the byte output.

If someone can take a look, I'm more than willing to assist!

**Comment by davidski** [comment]
Finally solved this one. It turns out that PowerShell converts all piped output via the console's encoding and there is no reliable encoding that won't munge binary output. There's some work being done to make this experience better in the core PowerShell environment over at https://github.com/PowerShell/PowerShell/issues/1908. In the meantime, using the process documented at http://www.leeholmes.com/blog/2007/01/18/redirecting-binary-output-in-powershell/ worked for getting a valid `apex build` output when running under PoSH.

Not an `apex` issue, though a version of the `build` command that did a file write instead of dumping the binary to stdout wouldn't make me unhappy. :wink:

tl;dr - Binary STDOUT is fraught with peril under modern Windows.
